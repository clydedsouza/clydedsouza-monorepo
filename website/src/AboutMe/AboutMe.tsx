function AboutMe() {
  return (
    <div>
      <h2>About me</h2>
      <p>
        Hi! I&apos;m Clyde D&apos;Souza. I&apos;m a creative and self-driven
        person who likes to code and do other things. I&apos;m currently working
        as a senior software engineer at
        <a
          href="https://www.xero.com/nz/?ref=clydedsouza.net"
          rel="noreferrer"
          target="_blank"
        >
          Xero
        </a>
        . Previously, I&apos;ve worked with
        <a
          href="https://www.datacom.co.nz/?ref=clydedsouza.net"
          rel="noreferrer"
          target="_blank"
        >
          Datacom
        </a>
        and
        <a
          href="https://www.heritagehotels.co.nz/?ref=clydedsouza.net"
          rel="noreferrer"
          target="_blank"
        >
          Heritage Hotels
        </a>
        .
      </p>
      <p>
        I was the event director of a non-profit educational event called Light
        & Spark NPO. I conceptualized and executed this event in Mumbai along
        with my
        <a
          href="https://lightandsparknpo.github.io/team/"
          rel="noreferrer"
          target="_blank"
        >
          amazing team
        </a>
        . Read more about it
        <a
          href="https://lightandsparknpo.github.io/2019-11-20-ruia-mumbai-2018/"
          rel="noreferrer"
          target="_blank"
        >
          here
        </a>
        .
      </p>
      <p>
        I&apos;ve published a book titled Mama, Tell Me a Story which is
        available on
        <a
          href="http://mamatellmeastory.clydedsouza.net/"
          rel="noreferrer"
          target="_blank"
        >
          these platforms
        </a>
        . It&apos;s a collection of twelve short bedtime stories that parents
        will love reading to their kids over and over again. Go buy it now!
      </p>
      <p>
        In my spare time, I like to explore other interests like
        <a href="https://medium.com/@clydedz" rel="noreferrer" target="_blank">
          writing
        </a>
        ,
        <a
          href="https://www.behance.net/clydedz"
          rel="noreferrer"
          target="_blank"
        >
          designing
        </a>
        ,
        <a
          href="https://www.skillshare.com/r/user/clydedsouza"
          rel="noreferrer"
          target="_blank"
        >
          teaching
        </a>
        , and I&apos;ve also created many
        <a href="/#/projects">other projects</a>.
      </p>
      <p>
        I&apos;m an alumnus of
        <a
          href="https://www.aut.ac.nz/?ref=clydedsouza"
          rel="noreferrer"
          target="_blank"
        >
          Auckland University of Technology (AUT)
        </a>
        with a Postgraduate Diploma in Computer and Information Sciences and
        holds a Bachelor of Science degree specializing in Information
        Technology from University of Mumbai, India.
      </p>
    </div>
  )
}

export default AboutMe
