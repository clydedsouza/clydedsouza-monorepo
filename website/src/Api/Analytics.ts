import mixpanel from 'mixpanel-browser'
import { PageTypes } from '../Components/Navigation/PageTypes'
import {
  ILinkClickedProps,
  MIXPANEL_DEV_PROJECT_ID,
  MIXPANEL_PROD_PROJECT_ID,
} from './IAnalytics'

export const initAnalyticsWithSuperProperties = () => {
  const mixpanelProjectId =
    process.env.NODE_ENV === 'production'
      ? MIXPANEL_PROD_PROJECT_ID
      : MIXPANEL_DEV_PROJECT_ID

  try {
    mixpanel.init(mixpanelProjectId)
    mixpanel.register({
      environment: process.env.NODE_ENV,
    })
  } catch {
    undefined
  }
}

export const sendPageViewEvent = (pageType: PageTypes) => {
  try {
    mixpanel.track('Page view', { page: pageType })
  } catch {
    undefined
  }
}

export const sendLinkClickedEvent = (props: ILinkClickedProps) => {
  try {
    mixpanel.track('Link clicked', {
      destination_url: props.link,
      link_type: props.type,
      page_location: props.location,
    })
  } catch {
    undefined
  }
}
