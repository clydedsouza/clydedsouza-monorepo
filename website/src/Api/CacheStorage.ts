import { PageTypes } from '../Components/Navigation/PageTypes'
import { IProjectData } from './IProjectData'

export const projectDataCacheMap: Map<PageTypes, IProjectData> = new Map()
