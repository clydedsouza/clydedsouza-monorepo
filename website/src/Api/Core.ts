import axios from 'axios'
import { PageTypes } from '../Types/PageTypes'
import { IProjectData } from '../Types/ProjectData'

const BASE_URL = 'https://api.clydedsouza.net/'

const getProjectApiUrl = (pageType: PageTypes) => {
  switch (pageType) {
    case PageTypes.Books:
      return `${BASE_URL}books.json`
    case PageTypes.Speaking:
      return `${BASE_URL}speaking.json`
    case PageTypes.Teaching:
      return `${BASE_URL}teaching.json`
    case PageTypes.Projects:
    default:
      return `${BASE_URL}pinned.json`
  }
}

export const getProjectData = (pageType: PageTypes): Promise<IProjectData> => {
  const request = getProjectApiUrl(pageType)
  return axios
    .get(request)
    .then((response) => {
      return response.data
    })
    .catch(() => {
      throw new Error('API error')
    })
}
