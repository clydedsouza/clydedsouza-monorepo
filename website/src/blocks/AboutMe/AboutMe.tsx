// import { useEffect } from "react";
// import { sendPageViewEvent } from '../../Api/Analytics'
import Bio from "./Bio/Bio";
import Highlights from './Highlights/Highlights'

function AboutMe() {
  // useEffect(() => {
  //   sendPageViewEvent(PageTypes.About);
  // }, []);

  return (
    <>
      <Bio />
      <br></br>
      <Highlights />
    </>
  );
}

export default AboutMe;
