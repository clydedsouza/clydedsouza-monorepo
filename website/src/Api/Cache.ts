import { PageTypes } from '../Types/PageTypes'
import { IProjectData } from '../Types/ProjectData'
import { getProjectData } from './Core'

const projectDataCacheMap: Map<PageTypes, Promise<IProjectData>> = new Map()

export const getCachedProjectData = (
  pageType: PageTypes
): Promise<IProjectData> => {
  if (!projectDataCacheMap.has(pageType)) {
    const getProjectDataPromise = getProjectData(pageType).catch((error) => {
      projectDataCacheMap.delete(pageType)
      return Promise.reject(error)
    })
    projectDataCacheMap.set(pageType, getProjectDataPromise)
  }

  return projectDataCacheMap.get(pageType)!
}
