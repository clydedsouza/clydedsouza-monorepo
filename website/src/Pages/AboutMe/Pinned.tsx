import { useEffect, useState } from 'react'
import { getCachedProjectData } from '../../Api/Cache'
import { PageTypes } from '../../Types/PageTypes'
import { IProject } from '../../Types/ProjectData'
import ProjectTile from '../Projects/ProjectTile'

function Pinned() {
  const [projectData, setProjectData] = useState<IProject[]>([])

  useEffect(() => {
    getCachedProjectData(PageTypes.Pinned)
      .then((apiResponse) => {
        setProjectData(apiResponse.data)
      })
      .catch(() => {
        setProjectData([])
      })
  }, [])

  return (
    <>
      <h2>Highlights</h2>
      <div id="pinnedProjects">
        <div id="pinnedProjectsList">
          {projectData.length > 0 ? (
            projectData.map((item) => (
              <ProjectTile {...item} key={item.title} />
            ))
          ) : (
            <p>Couldn&lsquo;t load the data at this stage</p>
          )}
        </div>
      </div>
    </>
  )
}

export default Pinned
