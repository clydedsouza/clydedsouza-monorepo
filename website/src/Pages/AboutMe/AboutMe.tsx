import Seo from '../../Components/Seo/Seo'
import Bio from './Bio'
import Pinned from './Pinned'

function AboutMe() {
  return (
    <>
      <Seo />
      <Bio />
      <Pinned />
    </>
  )
}

export default AboutMe
