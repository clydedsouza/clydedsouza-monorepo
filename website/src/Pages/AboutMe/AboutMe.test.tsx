import {
  render,
  screen,
  waitForElementToBeRemoved,
} from '@testing-library/react'
import { emptyProject } from '../../../TestHelpers/testHelper'
import * as apiCacheModule from '../../Api/Cache'
import AboutMe from './AboutMe'

jest.mock('../../Api/Cache')

const highlightsDataMockResponse = [
  {
    ...emptyProject,
    id: '123-amazing-alloy',
    title: 'Amazing alloy',
    date: '01 May 2020',
  },
  {
    ...emptyProject,
    id: '012-zinger-zebra',
    title: 'Zinger zebra',
    date: '01 Jul 2017',
  },
  {
    ...emptyProject,
    id: '020-madness-monkey',
    title: 'Madness monkey',
    date: '30 March 2023',
  },
]

describe('About me', () => {
  beforeEach(() => {
    jest
      .spyOn(apiCacheModule, 'getCachedProjectData')
      .mockImplementationOnce(() =>
        Promise.resolve({
          app: {},
          data: [...highlightsDataMockResponse],
        })
      )
  })

  it('should render about me', async () => {
    render(<AboutMe />)
    await waitForElementToBeRemoved(() => screen.queryByRole('progressbar'))

    expect(document.body).toMatchSnapshot()
  })
})
