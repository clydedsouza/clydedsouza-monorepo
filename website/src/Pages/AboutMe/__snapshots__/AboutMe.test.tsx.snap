// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About me should render about me 1`] = `
<body>
  <div>
    <h2>
      About me
    </h2>
    <p>
      Hi! I'm Clyde D'Souza. I'm a creative and self-driven person who likes to code and do other things. I'm currently working as a senior software engineer at
       
      <a
        href="https://www.xero.com/nz/?ref=clydedsouza.net"
        rel="noreferrer"
        target="_blank"
      >
        Xero
      </a>
      . Previously, I've worked with
       
      <a
        href="https://www.datacom.co.nz/?ref=clydedsouza.net"
        rel="noreferrer"
        target="_blank"
      >
        Datacom
      </a>
       
      and
       
      <a
        href="https://www.heritagehotels.co.nz/?ref=clydedsouza.net"
        rel="noreferrer"
        target="_blank"
      >
        Heritage Hotels
      </a>
      .
    </p>
    <p>
      I was the event director of a non-profit educational event called Light & Spark NPO. I conceptualized and executed this event in Mumbai along with my
       
      <a
        href="https://lightandsparknpo.github.io/team/"
        rel="noreferrer"
        target="_blank"
      >
        amazing team
      </a>
      . Read more about it
       
      <a
        href="https://lightandsparknpo.github.io/2019-11-20-ruia-mumbai-2018/"
        rel="noreferrer"
        target="_blank"
      >
        here
      </a>
      .
    </p>
    <p>
      I've published a book titled Mama, Tell Me a Story which is available on
       
      <a
        href="http://mamatellmeastory.clydedsouza.net/"
        rel="noreferrer"
        target="_blank"
      >
        these platforms
      </a>
      . It's a collection of twelve short bedtime stories that parents will love reading to their kids over and over again. Go buy it now!
    </p>
    <p>
      In my spare time, I like to explore other interests like
       
      <a
        href="https://medium.com/@clydedz"
        rel="noreferrer"
        target="_blank"
      >
        writing
      </a>
      ,
       
      <a
        href="https://www.behance.net/clydedz"
        rel="noreferrer"
        target="_blank"
      >
        designing
      </a>
      ,
       
      <a
        href="https://www.skillshare.com/r/user/clydedsouza"
        rel="noreferrer"
        target="_blank"
      >
        teaching
      </a>
      , and I've also created many
       
      <a
        href="/#/projects"
      >
        other projects
      </a>
      .
    </p>
    <p>
      I'm an alumnus of
       
      <a
        href="https://www.aut.ac.nz/?ref=clydedsouza"
        rel="noreferrer"
        target="_blank"
      >
        Auckland University of Technology (AUT)
      </a>
       
      with a Postgraduate Diploma in Computer and Information Sciences and holds a Bachelor of Science degree specializing in Information Technology from University of Mumbai, India.
    </p>
    <br />
    <h2>
      Highlights
    </h2>
    <div
      id="pinnedProjects"
    >
      <div
        id="pinnedProjectsList"
      >
        <div
          class="project-tile"
        >
          <div>
            <span>
              Graphic Design
            </span>
          </div>
          <div>
            <h3>
              Tulips cover page designs
            </h3>
            <p>
              I was responsible for designing the cover page for Tulips - a church bulletin. Each cover page was designed in accordance to a theme for that particular issue and conveyed the underlying meaning of the theme.
            </p>
            <div
              class="made-using"
            >
              <span>
                Photoshop
              </span>
            </div>
          </div>
        </div>
        <div
          class="project-tile"
        >
          <div>
            <span>
              Website
            </span>
          </div>
          <div>
            <h3>
              Profile Sticker
            </h3>
            <p>
              Profile Sticker is a web app that allows a user to add a sticker from different themes to their profile picture. The user can choose to upload a picture from and download to a computer or Facebook.
            </p>
            <div
              class="made-using"
            >
              <span>
                HTML
              </span>
              <span>
                CSS
              </span>
              <span>
                SCSS
              </span>
              <span>
                JavaScript
              </span>
              <span>
                Mustache.js
              </span>
            </div>
            <div
              class="project-links"
            >
              <a
                href="https://profilesticker.net"
                rel="noreferrer"
                target="_blank"
              >
                <i
                  class="fas fa-globe"
                />
              </a>
              <a
                href="https://github.com/profilesticker/profilesticker.github.io/"
                rel="noreferrer"
                target="_blank"
              >
                <i
                  class="fab fa-github"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
